### ISSUE TITLE
It shouldn't be possible to disconnect/delete a network with instances attached

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#422]
Originally reported of Google Code with ID 366.

```
The following commands will disconnect and then delete a network even if instances are attached to it:
# gnt-network disconnect --no-conflicts-check vlan978
# gnt-network remove vlan978

This should be impossible. Also the config should self-check for a network's non-existance if nics are associated to it and it should refusing saving it, as it's a broken situation.

Then we can remove "gracefully failing" around if the lookup of an instance's network doesn't exist.

Thanks,

Guido

```
Originally added on 2013-02-07 15:40:55 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This is fixed already (by me and Helga). See commits 6bb939a and 0ae4b35.

Thanks,
dimara
```
Originally added on 2013-02-07 15:48:45 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Well commits 6bb939a, 1cce2c4, 0ae4b35 are pushed only in master. Should
they cherry-picked in devel-2.7?

Thanks,
dimara
```
Originally added on 2013-02-08 15:22:45 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Yes, since they affect the current release they need to be pushed there.

I will cherry-pick them.

Thanks,
Guido

```
Originally added on 2013-02-11 14:40:04 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-02-11 15:10:00 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-02-11 15:10:53 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The commits were cherry-picked, as such I am marking this issue as fixed.

Thanks,

Guido
```
Originally added on 2013-02-26 16:04:34 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
